coi/src/embedding.rs

```diff
@@ -94,82 +69,25 @@ where
 ///
 /// See [`pairwise_cosine_similarity`] for details.
 pub fn cosine_similarity(a: ArrayView1<'_, f32>, b: ArrayView1<'_, f32>) -> f32 {
```

Comment:
🔧 we need to rename the function to reflect on the new body, maybe sth like
```suggestion
pub fn normalized_dot_product(a: ArrayView1<'_, f32>, b: ArrayView1<'_, f32>) -> f32 {
```
it's probably also better to merge #768 first to avoid conflicts with `pairwise_cosine_similarity()`.